src/main/java/TaskSafe.java

```diff
@@ -0,0 +1,105 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.PatternSyntaxException;
+
+/*------- Local imports --------*/
+import exceptions.DukeException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.TaskType;
+
+public class TaskSafe {
+
+    /*------------ PUBLIC STAIC VARIABLES --------- */
+    public static final String DATA_PATH = "/data/duke.txt";
+    public static String rootPath = new File(".").getAbsolutePath();
+
+    /*-------------- SAVE ------------ */
+    private static void appendToFile(String file_path, Task task) throws DukeException{
+        try {
+            FileWriter fw = new FileWriter(file_path,true);
+            char isDone = task.isDone() ? '1' : '0';
+            String taskToAdd =
+                    task.getTaskChar() + "|" + isDone + "|" + task.getFullDescription() + '\n';
+            fw.write(taskToAdd);
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e);
+            throw new DukeException("Failed to save following task: \n" + task);
+        }
+    }
+
+    /**
+     * Saves a tasklist to a file and overwrites what is in it
+     * @param taskList
+     */
+    public static void saveToFile(String file_path, ArrayList<Task> taskList) {
```

Comment:
Since file path is constant, perhaps it will be better for file path to be handled within the TaskSafe class and does not need to be a parameter for this method